### ISSUE TITLE
Main thread terminates early

### ISSUE THREAD

**Issue by mccallum-sgd** [MasterKazurel/AxiFi#4]
### Steps to replicate
1) Start the program
2) Enter admin login information
    - username: csadmin, password: CSCI323
3) Login

### Error
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at com.cyber.control.MasterControl.switchTo(MasterControl.java:33)
	at com.cyber.control.LoginControl.lambda$0(LoginControl.java:22)
```
### Problem
`ControlManager` is null at the [point of login](https://github.com/mccallum-sgd/AxiFi/blob/788f926e00228b98f8dd22f90ab5c152bf51e8d5/com/cyber/control/MasterControl.java#L33). This is called by the [`ActionListener`](https://github.com/mccallum-sgd/AxiFi/blob/788f926e00228b98f8dd22f90ab5c152bf51e8d5/com/cyber/control/LoginControl.java#L22) in `LoginControl` when the user logs in.

### Cause
The main thread is terminating right after the call to `master.start()` in `Driver`. This can only mean the JVM detects that `MasterControl` is unreachable (i.e. is never used again), and gets collected by GarbageCollection.

___
 I thought the `ActionListener`s in `LoginControl` would keep the main thread alive, but apparently not. 

My theory is that the JVM is creating new threads for the subcontrollers, and therefore keeping those running while the main thread exits because apparently the `MasterControl` object has nothing more to do (even though it has references to it's `ControlManager` that can still potentially be, and are, used).
(__EDIT: 10/24__ _it doesn't look like it's creating new threads, the main thread just isn't waiting for the `ActionListener`s._)

I can see all of this happening when I step through the code, but I'm not exactly sure why and I don't know what I can do differently to fix it.

**Comment by mccallum-sgd** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Overhaul](https://trello.com/c/QAgtwwkq/7-overhaul)

**Comment by mccallum-sgd** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Issues](https://trello.com/c/RXPXF3Jk/15-issues)

**Comment by mccallum-sgd** [comment]
Working on branch [issue-4](https://github.com/MasterKazurel/AxiFi/tree/issue-4).

**Comment by mccallum-sgd** [comment]
Fixed by 12cbd74418b847dd39e9e27a3f5c5f4c2a541f34.
